src/mlpack/methods/ann/layer/add_merge_impl.hpp

```diff
@@ -41,6 +41,17 @@ AddMerge<InputDataType, OutputDataType, CustomLayers...>::AddMerge(
   // Nothing to do here.
 }
 
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+AddMerge<InputDataType, OutputDataType, CustomLayers...>::AddMerge(
+    const AddMerge& layer) :
+    model(layer.model),
+    run(layer.run),
+    ownsLayers(layer.ownsLayers)
+{
+  // Nothing to do here.
```

Comment:
Still, It was failing I just removed that code.

Reply:
I do think that `network` needs to be copied though---what was the error?  Maybe some of the layers in `network` don't have the copy constructor implemented yet in the tests you are running?

Reply:
When we are using add_merge in BRNN it is not copying any matrix that's why returning this error.
```
error: addition: incompatible matrix dimensions: 0x0 and 7x1
terminate called after throwing an instance of 'std::logic_error'
  what():  addition: incompatible matrix dimensions: 0x0 and 7x1
```